openapiart/common.py

```diff
@@ -187,101 +187,118 @@ def _decode(self, dict_object):
+            return True
+        if ip[:2] == "::":
+            ip = ip.replace("::", "0:")
+        elif ip[-2:] == "::":
+            ip = ip.replace("::", ":0")
+        else:
+            ip = ip.replace("::", ":0:")
+        try:
+            return all([
+                True if (0 <= int(oct, 16) <= 65535) and (1 <= len(oct) <= 4) else False
+                for oct in ip.split(":")
+            ])
+        except Exception:
+            return False
 
     def validate_hex(self, hex):
-        if isinstance(hex, list):
-            return all([self._HEX_REGEX(h) for h in hex])
-        if self._HEX_REGEX.match(hex):
+        if hex is None or not isinstance(hex, str):
+            return False
+        try:
+            int(hex, 16)
             return True
-        return False
+        except Exception:
+            return False
 
     def validate_integer(self, value):
-        if not isinstance(value, list):
-            value = [value]
-        return all([isinstance(i, int) for i in value])
+        return isinstance(value, int)
 
     def validate_float(self, value):
-        if not isinstance(value, list):
-            value = [value]
-        return all([isinstance(i, (float, int)) for i in value])
+        return isinstance(value, (int, float))
 
     def validate_string(self, value):
-        if not isinstance(value, list):
-            value = [value]
-        return all([isinstance(i, (str, unicode)) for i in value])
-
+        return isinstance(value, (str, unicode))
+    
     def validate_bool(self, value):
-        if not isinstance(value, list):
-            value = [value]
-        return all([isinstance(i, bool) for i in value])
-
-    def validate_list(self, value):
-        # TODO pending item validation
-        return isinstance(value, list)
+        return isinstance(value, bool)
 
+    def validate_list(self, value, itemtype):
+        if value is None or not isinstance(value, list):
+            return False
+        v_obj = getattr(self, "validate_{}".format(itemtype), None)
+        if v_obj is None:
+            raise AttributeError("{} is not a valid attribute".format(itemtype))
+        return [v_obj(item) for item in value]
+    
     def validate_binary(self, value):
-        if not isinstance(value, list):
-            value = [value]
-        return all([all([True if int(b) == 0 or int(b) == 1 else False for b in binary]) for binary in value])
-
-    def types_validation(self, value, type_, err_msg):
-        type_map = {int: "integer", str: "string", float: "float", bool: "bool", list: "list", "int64": "integer", "int32": "integer", "double": "float"}
+        if value is None or not isinstance(value, str):
+            return False
+        return all([
+            True if int(bin) == 0 or int(bin) == 1 else False
+            for bin in value
+        ])
+    
+    def types_validation(self, value, type_, err_msg, itemtype=None):
+        type_map = {
+            int: "integer", str: "string",
```

Comment:
list issue still present (https://github.com/open-traffic-generator/openapiart/issues/43)
I am getting error at the time of run this 
```python
    d = cfg.devices.device(name='d', container_name=p1.name)[-1]
    eth = d.ethernet
    eth.name = 'e'
    eth.mac = ['00:01:00:00:00:01']
```
```python
>           raise TypeError(err_msg)
E           TypeError: Invalid ['00:01:00:00:00:01'] format, expected mac at <class 'snappi.snappi.DeviceEthernet'>
```

Reply:
I think you cannot pass list to eth.mac right?